src/common/offline_transformations/src/pruning/init_masks.cpp

```diff
@@ -75,8 +76,34 @@ class ngraph::pass::init_masks::InitMatMulMask : public MatcherPass {
             NodeVector weights_calculation_nodes;
             auto cur_node = matmul->get_input_node_shared_ptr(1);
 
+            if (cur_node->get_output_partial_shape(0).is_dynamic()) return false;
+            auto dim_order = std::vector<int64_t>();
+            const auto input_size = cur_node->get_output_shape(0).size();
+            for (int64_t idx = 0; idx < input_size; ++idx)
+                dim_order.push_back(idx);
+
             while (!ngraph::is_type<opset6::Constant>(cur_node) && cur_node->inputs().size()) {
                 weights_calculation_nodes.push_back(cur_node);
+                if (ngraph::is_type<opset6::Transpose>(cur_node)) {
+                    const auto forward_order = get_constant_from_source(cur_node->get_input_node_shared_ptr(1));
+                    if (!forward_order) return false;
+                    const auto forward_order_vec = forward_order->cast_vector<int64_t>();
+                    if (forward_order_vec.size() != input_size) return false;
+                    auto new_order = std::vector<int64_t>();
+                    for (size_t i = 0; i < forward_order_vec.size(); ++i) {
+                        const auto dim = std::find(forward_order_vec.begin(), forward_order_vec.end(), i) - forward_order_vec.begin();
+                        // Dim should be valid because of transpose operation input_order input restrictions
+                        new_order.push_back(dim_order[dim]);
+                    }
```

Comment:
I guess we can simplify this and re-write without find, please check this
```suggestion
                    auto new_order = std::vector<int64_t>(forward_order_vec.size());
                    for (size_t i = 0; i < forward_order_vec.size(); ++i) {
                        new_order[forward_order_vec[i]] = dim_order[i];
                    }
```

Reply:
You are right! Thanks!